code/modules/shuttles/marine_ferry.dm

```diff
@@ -467,14 +467,8 @@
 	SEND_GLOBAL_SIGNAL(COMSIG_GLOB_HIJACK_IMPACTED)
 
 	// Sleep while the explosions do their job
-	var/explosion_alive = TRUE
-	while(explosion_alive)
-		explosion_alive = FALSE
-		for(var/datum/automata_cell/explosion/E in GLOB.cellauto_cells)
-			if(E.explosion_cause_data && E.explosion_cause_data.cause_name == "dropship crash")
-				explosion_alive = TRUE
-				break
-		sleep(1)
+	while(SSexplosions.is_exploding())
+		sleep(10)
```

Comment:
sleep(world.tick_lag)
or UNTIL() if you're fine with stoplag()

Reply:
huh